roles/zabbix_server/tasks/initialize-mysql.yml

```diff
@@ -0,0 +1,149 @@
+---
+# task file for mysql
+- name: "Install MySQL dependencies"
+  ansible.builtin.package:
+    name: "{{ _zabbix_server_mysql_dependencies | select | list }}"
+  environment:
+    http_proxy: "{{ zabbix_http_proxy | default(None) | default(omit) }}"
+    https_proxy: "{{ zabbix_https_proxy | default(None) | default(omit) }}"
+  register: _zabbix_server_dependencies_installed
+  until: _zabbix_server_dependencies_installed is succeeded
+  become: true
+  tags:
+    - install
+    - database
+    - dependencies
+
+- name: "MySQL Database prep"
+  when: zabbix_server_database_creation
+  delegate_to: "{{ zabbix_server_real_dbhost | default(zabbix_server_dbhost_run_install | ternary(delegated_dbhost, inventory_hostname)) }}"
+  vars:
+    delegated_dbhost: "{{ (zabbix_server_dbhost == 'localhost') | ternary(inventory_hostname, zabbix_server_dbhost) }}"
+  tags:
+    - database
```

Comment:
Yes, until I started using ansible-lint, I also always put the `when`, `delegate_to`, `vars` and such at the bottom. ansible-lint had a pretty good argument for putting these things before `block/rescue/always`. Since then (even though it was a bit jarring at first) I've started putting `when` right after `name` for many non-block tasks aswell. It forces you to write a better description for `name` and you can immediately read the condition.

> "Initialize database, when zabbix_server_database_creation

or

> "Install packages, when item.required"

It doesn't feel right for every type of tast though. And again, I know it looks and feels weird at first, but if you give it a chance, it'll grow on you :smile_cat: